src/include/distributed/intermediate_result_pruning.h

```diff
@@ -13,6 +13,15 @@
 
 #include "distributed/subplan_execution.h"
 
+/*
+ * TODO: In theory, a node can have the nodeID UINT32_MAX,
+ * but in practice that doesn't sound realistic. In addition
+ * to that, pg_dist_node_nodeid_seq would error out after
+ * getting to UINT32_MAX as that's the limit. Thus, we'd have
+ * a bigger problem than this.
+ */
+#define LOCAL_NODE_ID UINT32_MAX
```

Comment:
Can we change the sequence that is used to assign nodeId so that UINT32_MAX will never be assigned and causes an error earlier? I guess we can deduct the end of the range by one during the citus upgrade script.

Reply:
yeah, I was lazy to do it :) see the latest commit, and let me know if that looks good now?